src/Time/Units.hs

```diff
@@ -120,6 +122,21 @@ type Fortnight   = 1209600 :% 1
 newtype Time (rat :: Rat) = Time { unTime :: RatioNat }
     deriving (Eq, Ord, Enum, Real, RealFrac, Generic)
 
+-- | Addition is associative binary operation for 'Semigroup' of 'Time'.
+instance Semigroup (Time (rat :: Rat)) where
+    (<>) = (+)
+    {-# INLINE (<>) #-}
+    sconcat = foldl' (<>) mempty
+    {-# INLINE sconcat #-}
```

Comment:
OMG, I'm stupid. `stimes` also can be implemented efficiently using multiplication. Checkout `Semigroup` instance for `Sum`:

* http://hackage.haskell.org/package/base-4.10.1.0/docs/src/Data.Semigroup.html#line-243